RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc

```diff
@@ -0,0 +1,1038 @@
+		    }//for proto_seg                                                                                                                                                                 
+		  }//for rec_hits   
+		}//else 
+	      }//proto seg not empty 
+	    }
+	  }  //   h1 & h2 close 
+	  if (segok)  
+            break; 
+	}  //  i2 
+      }  //  i1 
+
+      //add the reconstructed segments 
+      for(int j = 0;j < common_it+1; j++){ 
+
+	proto_segment = best_proto_segment[j]; 
+	best_proto_segment[j].clear(); 
+
+	//SKIP empty proto-segments                                                                                                                                                
+        if(proto_segment.size() == 0) continue;
+
+	updateParameters();                                                                                                                                 
+
+	// Create an actual CSCSegment - retrieve all info from the fit
+	CSCSegment temp(sfit_->hits(), sfit_->intercept(),
+                       sfit_->localdir(), sfit_->covarianceMatrix(), sfit_->chi2());
+	delete sfit_;
+	sfit_ = 0;
+	segments.push_back(temp);                         
+	//if the segment has 3 hits flag them as used in a particular way 
+	if(proto_segment.size() == 3){ 
+	  flagHitsAsUsed(rechits, used3p); 
+	} 
+	else{ 
+	  flagHitsAsUsed(rechits, used);   
+	} 
+	proto_segment.clear(); 
+      }  
+    }//for n_seg_min 
+
+    if(!doCollisions && search_disp){
+
+      //reset params and flags for the next chamber                                                                                                                           
+      search_disp = false;
+      doCollisions = true;
+      dRMax = 2.0;
+      dPhiMax = dPhiMax/2;
+      dRIntMax = dRIntMax/2;
+      dPhiIntMax = dPhiIntMax/2;
+      chi2Norm_2D_ = chi2Norm_2D_/5;
+      chi2_str_ = 100;
+      chi2Max = chi2Max/2;
+    }     
+
+    std::vector<CSCSegment>::iterator it =segments.begin(); 
+    bool good_segs = false; 
+    while(it != segments.end()) { 
+      if ((*it).nRecHits() > 3){ 
+	good_segs = true; 
+	break; 
+      } 
+      ++it;	 
+    }     
+
+    if (good_segs) {  // only change window if not enough good segments were found (bool can be changed to int if a >0 number of good segs is required) 
+      search_disp = true;
+      continue;//proceed to search the segs from displaced vertices
+    }
+
+    // Increase cut windows by factor of wideSeg only for collisions 
+    if(!doCollisions && !search_disp) break;     
+    windowScale = wideSeg; 
+  }  //  ipass 
+
+  //get rid of enchansed 3p segments 
+  std::vector<CSCSegment>::iterator it =segments.begin(); 
+  while(it != segments.end()) { 
+    if((*it).nRecHits() == 3){ 
+      bool found_common = false; 
+      std::vector<CSCRecHit2D> theseRH = (*it).specificRecHits(); 

```

Comment:
specificRecHits() returns a reference.
I don't see any modifications to ``` theseRH ``` in this scope. 
Please change to ``` const std::vector<CSCRecHit2D>& theseRH = ```